dashboard/serializers.py

```diff
@@ -25,6 +25,25 @@ def serialize_enrollments(cls, enrollments):
         return [{'title': course.title} for course in enrolled_courses]
 
     @classmethod
+    def count_courses_passed(cls, program, mmtrack):
+        """
+        Returns number of passed courses.
+
+        Args:
+            program (Program): A program object.
+            mmtrack (dashboard.utils.MMTrack): a instance of all user information about a program
+        Returns:
+            num_complete_courses: A number of passed courses.
+        """
+        num_courses_passed = 0
+        for course in program.course_set.all():
+            course_run = course.first_unexpired_run()
+            if course_run and mmtrack.has_passed_course(course_run.edx_course_key):
+                num_courses_passed += 1
+
+        return num_courses_passed
```

Comment:
i don't think the logic for this method is correct. for each run, we're running `course.first_unexpired_run()` to get a course run and check if it has passed. if a user has passed a course, that run is in the past, or in other words, it has expired. 

also, `MMTrack` already populates a list of every course run key in a given program (https://github.com/mitodl/micromasters/blob/master/dashboard/utils.py#L57), so if we need to know the number of passed course runs, that method can be implemented in `MMTrack` with very little work. if we need to know the number of _courses_ passed, we can implement that in `MMTrack` as well and avoid extra queries. we can change this line (which i linked above):

```python
self.course_ids = set(CourseRun.objects.filter(course__program=self.program).exclude(
                Q(edx_course_key__isnull=True) | Q(edx_course_key__exact='')
            ).values_list("edx_course_key", flat=True))
```

and instead implement it like this:

```python
# Maps a CourseRun's edx_course_key to its parent Course id
self.edx_key_course_map = dict(
    CourseRun.objects.filter(course__program=program).exclude(
        Q(edx_course_key__isnull=True) | Q(edx_course_key__exact='')
    ).values_list("edx_course_key", "course__id")
)
self.course_ids = self.edx_key_course_map.keys()
```

we can then implement `num_courses_passed` in `MMTrack` by looping through `self.course_ids`, checking `self.has_passed_course(course_id)`, and getting the count of unique course id's in `self.edx_key_course_map`. @noisecapella what do you think?